src/integrations/bsuir/mapper.ts

```diff
@@ -0,0 +1,168 @@
+import { GroupScheduleResponse, WeekDayScheduleResponse } from './client';
+import { compareDesc } from 'date-fns';
+import { addDays } from 'date-fns';
+import { getDay } from 'date-fns';
+import { getDate } from 'date-fns';
+import { getMonth } from 'date-fns';
+import { getYear } from 'date-fns';
+import { getWeek } from 'date-fns';
+import { formatISO } from 'date-fns';
+import { differenceInMinutes } from 'date-fns';
+import { Lesson } from '../models';
+
+export class BSUIRResponseMapper {
+  public getSchedule(groupSchedule: GroupScheduleResponse): (Lesson | Lesson[])[] {
```

Comment:
Why return type is not deterministic? Can you update your algorithm to simply return Lesson[]?